python/triton/runtime/backends/hip.c

```diff
@@ -96,17 +96,24 @@ static PyObject *loadBinary(PyObject *self, PyObject *args) {
                     (void *)(uintptr_t)logbufsize, (void *)_log, (void *)1};
 
   // launch HIP Binary
+  int32_t n_regs = 0;
+  int32_t n_spills = 0;
   hipModule_t mod;
   hipFunction_t fun;
   Py_BEGIN_ALLOW_THREADS;
   HIP_CHECK_AND_RETURN_NULL_ALLOW_THREADS(hipModuleLoadDataEx(&mod, hsaco, 5, opt, optval));
   HIP_CHECK_AND_RETURN_NULL_ALLOW_THREADS(hipModuleGetFunction(&fun, mod, name));
+  HIP_CHECK_AND_RETURN_NULL_ALLOW_THREADS(
+      hipFuncGetAttribute(&n_regs, HIP_FUNC_ATTRIBUTE_NUM_REGS, fun));
+  HIP_CHECK_AND_RETURN_NULL_ALLOW_THREADS(
+      hipFuncGetAttribute(&n_spills, HIP_FUNC_ATTRIBUTE_LOCAL_SIZE_BYTES, fun));
+  // somehow ISA dumping seems vgpr_spill_count always need minus 1
+  if(n_spills != 0) n_spills = n_spills / 4 - 1;
```

Comment:
Than why does it need to / 4? 4 SIMDs?

Reply:
the n_spills get from 109 is number of bytes, my understanding is dividing by 4 converts the total amount of local memory in bytes to the equivalent number of 4-byte variables. It can be a way to estimate how many such variables are being spilled to local memory from registers, given variables are often 4 bytes in size (this is typical for int or float types in C/C++), to confirm this, we may need find out how the vgpr_spill_count is computed, this is currently matching with vgpr_spill_count in the amdgcn dump